### PR TITLE
handle unmarked duplicates with mate MQ = 0 in Mutect2

### PR DESCRIPTION
**Pull request by davidbenjamin**
@LeeTL1220 This is for Sachet's artifact, ready to review pending the results on Firecloud looking good.

### CODE REVIEW

**Comment by LeeTL1220** [comment]
@davidbenjamin Results on FC looked really good.  Reviewed manually.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java [hunk 1]
```diff
@@ -260,6 +263,30 @@ public void writeHeader(final VariantContextWriter vcfWriter, final Set<VCFHeade
         }
     }
 
+    private void removeReadStubs(final AssemblyRegion assemblyRegion) {
+        final List<GATKRead> readStubs = assemblyRegion.getReads().stream()
+                .filter(r -> r.getLength() < AssemblyBasedCallerUtils.MINIMUM_READ_LENGTH_AFTER_TRIMMING).collect(Collectors.toList());
+        assemblyRegion.removeAll(readStubs);
+    }
+
+    private void removeUnmarkedDuplicates(final AssemblyRegion assemblyRegion) {
```
Comment (LeeTL1220) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java]: This looks like it considers:

- SVs:  This code checks whether the mates are clustered, even if on a different contig from the supporting reads of the variant.
- Duplicate start and end positions of supporting reads.

Please tell me if I have misinterpreted.  If I have not misinterpreted, then no action required.

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2IntegrationTest.java [hunk 2]
```diff
@@ -1150,4 +1150,20 @@ private static String keyForVariant( final VariantContext variant ) {
         return String.format("%s:%d-%d %s, %s", variant.getContig(), variant.getStart(), variant.getEnd(), variant.getReference(),
                 variant.getAlternateAlleles().stream().map(Allele::getDisplayString).sorted().collect(Collectors.toList()));
     }
+
+    @Test
+    public void testSachet() throws Exception {
```
Comment (LeeTL1220) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2IntegrationTest.java]: Can we rename to something about missed duplicates?  And if you want to thank Sachet in a comment, that should be okay.

**Comment by LeeTL1220** [comment]
@davidbenjamin Back to you

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java (hunk 1 above)
Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java]: That's right.

**Pull request merged**